pkg/volume/flexvolume/probe.go

```diff
@@ -228,6 +228,10 @@ func (prober *flexVolumeProber) updateEventsMap(eventDirAbs string, op volume.Pr
 // Each file or directory change triggers two events: one from the watch on itself, another from the watch
 // on its parent directory.
 func (prober *flexVolumeProber) addWatchRecursive(filename string) error {
+	// this may be called with an actual absolute path on Windows (with a C:\ prefix).
+	// But the prober.fs.Walk below will execute filepath.Join(fs.root, filenameAbove), which
+	// will result in an incorrect path, you can't join C:\path and C:\another\path.
+	filename = strings.TrimPrefix(filename, `C:\`)
```

Comment:
I know very little about go on Windows, this looks very odd to me. What if `prober.pluginDir` is on `D:\`? 

I think it's better to fix `DefaultFs.prefix()` (or `NewTempFs()`?) to count with Windows and join the paths correctly.

Reply:
Will look more into it. I've split this off into a separate PR: https://github.com/kubernetes/kubernetes/pull/116627